src/bower.json

```diff
@@ -0,0 +1,12 @@
+{
+  "name": "server",
+  "main": "index.html",
+  "dependencies": {
+    "polymer": "Polymer/polymer#^2.0.0",
```

Comment:
Is this Polymer loaded via an iframe and hence doesn't need to match the polymer on the docs site?

If so, does that mean polymer is loaded twice for a docs site user?

Reply:
To the first question, yes. A user can type Polymer 2 or hybrid syntax into the editor, even though the docs site uses Polymer 1 at the moment. (This is the case because we're looking at revamping the Polymer 2 Quick Tour using this element.)

To the second, also yes - @e111077 is there a way to avoid this?

Reply:
Only solution I can imagine is to check if the correct version of polymer is outside of the iframe and have it passed in. I don't agree that this is necessary

Reply:
Sorry, I'm not talking about the demo frame. I think that's actually a third instance of Polymer.
`<editor-server>` loads Polymer. All the elements in https://github.com/PolymerLabs/polymer-live-editor load Polymer too. Are any of them overlapped or are they all distinct?

Reply:
Thanks for the clarification. Here's a list of where I see Polymer being loaded:

`codemirror-textarea` loads Polymer because it's meant to be its own element that one could `bower install`.  (However, I'm now noticing that the `bower.json` in `codemirror-textarea` should be using hybrid syntax and it isn't.) 

Each `polymer-live-editor-file` imports Polymer. Then `polymer-live-editor` imports `polymer-live-editor-file`. The bower.json allows 1.9-2. 

Then `polymer-live-editor-server` loads Polymer 2.

Is there any way I can reduce the number of loads? It does seem like a lot of imports for one element. Thanks!

Reply:
Ah you are correct. The client loads polymer for the `live-editor` (shared with their instance of polymer if they are using it on their domain where they include this element e.g. polymer-project.org). Then this server loads an instance of Polymer in [`editor-server`](https://github.com/PolymerLabs/polymer-live-editor-server/blob/eb4b40fe43bd860c1acff6faa1dd73e4f0273893/src/editor-server/editor-server.html) in an iframe to this domain, and then the `editor-server` creates an iframe with the same domain that refreshes on edit and may load polymer if they use an html import to polymer.

I think the only avoidable load of polymer is in `editor-server`; if we are able to make it without using Polymer. Do you think it's worth it?

Additionally, do you think it would be worth it introducing a time-based debounce on recreating the iframe on edit?

Reply:
@eblaine clarification, each live editor file imports polymer, but that instance is shared across all files since they are all in the same project (thus the browser resolves the url).

Reply:
@e111077 i suppose there's no reason editor-server.html *needs* to know what Polymer is, since it basically exists to do socket.io stuff and call document.write(). However, this repo (`polymer-live-editor-server`) will still need Polymer as a dependency.

That said, I imagine most of the starter code blocks would HTML import `polymer-element` for learning purposes, so it might make sense for editor-server.html to load it anyway to prevent re-loading it all the time. What do you think?

Reply:
So if you have an HTML import to something and then do an html import inside of an iframe with no `src` defined, does it actually transmit bytes, or is it already resolved? I believe that it still xmits bytes, but I'm not sure. Can you check in Chrome dev tools since you have an instance running?

Reply:
Yes, you're right - looks like it still transmits bytes. So I suppose that means I should (1) keep polymer as a dependency in bower.json and (2) remove polymer from editor-server.html - is this correct?

Reply:
@eblaine that sounds good to me!